### PR TITLE
#39 Add dimensional options to keep

### PR DESCRIPTION
**Pull request by omidraha**
#39 Add dimensional options to keep

### CODE REVIEW

**Comment by jarun** [comment]
CI fails with:

    imgp:281:4: R1705: Unnecessary "elif" after "return" (no-else-return)
Please fix.

File: README.md [hunk 1]
```diff
@@ -164,7 +164,10 @@ optional arguments:
   -H, --hidden          include hidden (dot) files [default: off]
   -i, --includeimgp     re-process _IMGP files. * RISKY: refer to docs
   -k, --keep            skip (honors -c or --pr) images matching specified
```
Comment (jarun) [inline README.md:166]: Please indicate optional parameter correctly. Currently it appears the option doesn't take any parameter.

File: README.md [hunk 2]
```diff
@@ -164,7 +164,10 @@ optional arguments:
...
-                        hres or vres or --res=100 [default: off]
+                        hres or vres or --res=100 or
```
Comment (jarun) [inline README.md:167]: I can't remember what --res=100 here means. Can you please try to figure it out?

File: README.md [hunk 3]
```diff
@@ -164,7 +164,10 @@ optional arguments:
...
+                        -k res: skip res matching that is less than or equal to hres/vres or
+                        -k hresxvres: skip hres/vres matching that are less than or equal hres and vres
```
Comment (jarun) [inline README.md:169]: skip images that are less than or equal to hres and vres

File: imgp [hunk 4]
```diff
@@ -275,6 +277,10 @@ def resize_image(src):
 
         if convert and _format != 'PNG':
             return
+
```
Comment (jarun) [inline imgp:280]: No unnecessary newlines for each if-else please. At least look at the existing code.

File: imgp [hunk 5]
```diff
@@ -275,6 +277,10 @@ def resize_image(src):
...
+    elif keep and (hres <= khres or vres <= kvres):
```
Comment (jarun) [inline imgp:281]: Should this be at line 274?

Comment (jarun) [reply imgp:281]: if keep and (scale == 100 or hres <= khres or vres <= kvres)

File: imgp [hunk 6]
```diff
@@ -585,7 +595,9 @@ def main():
     eraseexif = args.eraseexif
     force = args.force
     includeimgp = args.includeimgp
-    keep = args.keep
+    if args.keep:
+        keep = True
+        khres, kvres = getres(args.keep) if 'x' in args.keep else getres('{}x{}'.format(args.keep, args.keep))
```
Comment (jarun) [inline imgp:600]: No no value if passed for `keep` set khres = HRES and kvres = VRES.

**Review by jarun — CHANGES_REQUESTED** [review]

File: imgp [hunk 7]
```diff
@@ -585,7 +603,14 @@ def main():
...
     force = args.force
     includeimgp = args.includeimgp
-    keep = args.keep
+    if args.keep:
+        keep = True
+        if 'x' in args.keep:
+            khres, kvres = getres(args.keep)
+        else:
+            khres = kvres = try_int(args.keep)
+    else:
```
Comment (jarun) [inline imgp:612]: No.

I meant if the call is:

    imgp -k
    // -k without any arguments

khres should be set to HRES and kvres should be set to VRES.

File: imgp [hunk 8]
```diff
@@ -60,6 +62,23 @@ count = 0  # successful conversion count
...
+    123
+    >>> try_int('ab30')
+    0
+    >>> try_int('abc')
+    0
+    """
+    try:
+        return int(value)
+    except ValueError:
+        return 0
```
Comment (jarun) [inline imgp:79]: We should gracefully exit the program in this case. it's not our job to fix user input.

File: README.md [hunk 9]
```diff
@@ -163,8 +163,11 @@ optional arguments:
   -f, --force           force to exact specified resolution [default: off]
   -H, --hidden          include hidden (dot) files [default: off]
   -i, --includeimgp     re-process _IMGP files. * RISKY: refer to docs
-  -k, --keep            skip (honors -c or --pr) images matching specified
-                        hres or vres or --res=100 [default: off]
+  -k, --keep [res|hresxvres]
```
Comment (jarun) [inline README.md:166]:
```
  -k, --keep [pixels|HxV]
                        - no args: skip (honors -c or --pr) images with hres/vres <= --res or --res <= 100
                        - pixels: skip images with H/V <= pixels
                        - HxV: skip images with hres <= H or vres <= V
                        [default: off]
```

File: imgp [hunk 10]
```diff
@@ -269,7 +284,7 @@ def resize_image(src):
 
     antialias = png_ip if _format == 'PNG' else PIL.Image.ANTIALIAS
 
-    if keep and (HRES == hres or VRES == vres or scale == 100):
+    if keep and (scale == 100 or hres <= khres or vres <= kvres):
```
Comment (jarun) [inline imgp:287]: `scale <= 100)`

File: imgp [hunk 11]
```diff
@@ -498,9 +513,12 @@ def parse_args(args=None, namespace=None):
            help='include hidden (dot) files [default: off]')
     addarg('-i', '--includeimgp', action='store_true',
            help='re-process _IMGP files. * RISKY: refer to docs')
-    addarg('-k', '--keep', action='store_true',
-           help='skip (honors -c or --pr) images matching specified \
-                 hres or vres or --res=100 [default: off]')
+    addarg('-k', '--keep', nargs='?', metavar='keep', default=False,
+           help='-k: skip (honors -c or --pr) images matching specified hres or vres or --res=100, \
```
Comment (jarun) [inline imgp:517]: help should be:

```
                        - no args: skip (honors -c or --pr) images with hres/vres <= --res or --res <= 100
                        - pixels: skip images with H/V <= pixels
                        - HxV: skip images with hres <= H or vres <= V
                        [default: off]
```

**Review by jarun — CHANGES_REQUESTED** [review]

**Comment by omidraha** [comment]
Sorry I'm busy ;)
Please fix yourself,
Thanks.

**Comment by jarun** [comment]
No problem. The code is not up to the mark so far. Anyway, I'll take care of it.

**Comment by jarun** [comment]
I don't think we need this per current handling. Explaining in the defect.

**Pull request closed without merging**